src/app/profile/update/update-guard.component.ts

```diff
@@ -0,0 +1,59 @@
+import { Injectable, OnDestroy } from '@angular/core';
+import { ActivatedRoute, CanActivate, Router } from '@angular/router';
+
+import { Observable, Subscription } from 'rxjs';
+
+import { AuthenticationService, UserService, User } from 'ngx-login-client';
+
+import { LoginService } from '../../shared/login.service';
+
+@Injectable()
+export class UpdateGuard implements CanActivate, OnDestroy {
+  subscription: Subscription;
+  activateObservable: Observable<boolean>;
+  differentUsername = true;
+  username = '';
+
+  constructor(
+    private route: ActivatedRoute,
+    private authService: AuthenticationService,
+    private router: Router,
+    private loginService: LoginService,
+    private userService: UserService
+  ) {
+    this.activateObservable = Observable.create(observer => {
+      this.subscription = this.userService.loggedInUser.subscribe(user => {
+        this.handleUserNotification(user);
+        if (this.differentUsername) {
+          this.router.navigate(['/', this.username]);
+        }
+        observer.next(!this.differentUsername);
+      });
+    });
+  }
+
+  handleUserNotification(user: User): void {
+    if (user && user.attributes && user.attributes.username) {
+      this.username = user.attributes.username;
+
+      // TODO: Right now it's a hardcoded link, so can't grab user (like params['user']).
+      this.router.url.split('/').forEach(urlToken => {
+        if (urlToken === this.username) {
+          this.differentUsername = false;
```

Comment:
You should change this function to return a boolean instead of using a temporary variable that is only used in the line immediately after this is called on line 27